core/src/elements/ons-carousel.js

```diff
@@ -33,7 +33,8 @@ const VerticalModeTrait = {
 
   _getElementSize: function() {
     if (!this._currentElementSize) {
-      this._currentElementSize = this.getBoundingClientRect().height;
+      var styling = getComputedStyle(this, null);
+      this._currentElementSize = this.getBoundingClientRect().height - parseInt(styling.getPropertyValue('border-right-width')) - parseInt(styling.getPropertyValue('border-left-width'));
```

Comment:
oooops, you're partially right, it should be `border-top-width` and `border-bottom-width`